src/components/campaignguide/ScenarioView.tsx

```diff
@@ -159,6 +160,38 @@ function ScenarioView({ componentId, campaignId, showLinkedScenario, processedSc
       },
     });
   }, [componentId, campaignId, processedScenario.id]);
+  
+  useEffect(() => {
+    const queue: NarrationTrack[] = [];
+    for (const scenarioStep of processedScenario.steps) {
+      if (scenarioStep.step.type === "resolution") {
+        const narration = processedScenario.scenarioGuide.resolution(
+          scenarioStep.step.resolution
+        )?.narration;
+        if (!narration) continue;
+
+        queue.push({
+          ...narration,
+          campaignCode: processedScenario.scenarioGuide.campaignGuide.campaignCycleCode(),
+          campaignName: processedScenario.scenarioGuide.campaignGuide.campaignName(),
+          scenarioName: processedScenario.scenarioGuide.scenarioName(),
+        });
+      } else if (scenarioStep.step.type === "story") {
+        const narration = scenarioStep.step.narration;
+        if (!narration) continue;
+
+        queue.push({
+          ...narration,
+          campaignCode: processedScenario.scenarioGuide.campaignGuide.campaignCycleCode(),
+          campaignName: processedScenario.scenarioGuide.campaignGuide.campaignName(),
+          scenarioName: processedScenario.scenarioGuide.scenarioName(),
+        });
+      }
+    }
+    console.log('here');
+    queueNarration(queue);
+  }, [processedScenario]);
```

Comment:
so `processedScenario` is going to change every time a decision is made during scenario setup (for every step of the scenario) -- it basically does a new-calculation from the top every time.

Is that handled in the `queueNarration` logic?

Reply:
Also the whole block should probably be guarded in a hasDissonantVoices (inside the useEffect, with `hasDissonantVoices` redux state in the dependencies.

Reply:
>Is that handled in the queueNarration logic?
Yep.